enhancements/sig-architecture/34-work-executor-group/README.md

```diff
@@ -0,0 +1,211 @@
+SAR requests to the spoke cluster, but this should be fine b/c the work-agent 
+is working asynchronously. Similar to "ServiceAccount", "Username" type will 
+make the work agent check whether that username has sufficient RBAC permission
+for manipulating the corresponding objects.
+
+
+Note that the authorization via subject access review api doesn't require the
+service account to be actually present in the managed cluster, so in the cases
+where you have too many executor subject to manage, the number of service 
+accounts resources in the managed cluster is bounded.
+
+A sample of the requesting `SubjectAccessReview` payload is:
+
+```yaml
+apiVersion: authentication.open-cluster-management.io/v1alpha1
+kind: SubjectAccessReview
+spec:
+  # requester is the referenced service-account in executor subject
+  user: "system:serviceaccount:<sa namespace>:<sa name>"
+  groups: ["system:serviceaccounts", "system:authenticated"]
+  # requesting resource attributes
+  resourceAttributes:
+    # verb
+    verb: <the intended verb, can be create/update/patch/delete>
+    # the GVR tuple
+    group: <api group>
+    version: <api version>
+    resource: <api resource name>
+    # the object identifier
+    name: <resource name>
+    namespace: <resource namespace if applicable>
+```
+
+### Architecture
+
+The overall architecture is shown in the picture below:
+
+![Architecture](./arch.png "Architecture")
+
+#### Workflow
+
+The following is what will happen when a `ManifestWork` with non-nil executor
+subject specified is applied to the hub cluster:
+
+0. Create a "ManifestWork" with non-nil executor subject
+1. The work-agent observes the new resource via watch event
+2. The work-agent builds a SubjectAccessReview request which:
+   - The target subject is the executor subject from `ManifestWork`
+   - The target attribute should contain the "*" verbs for all the inlined
+     objects in the spec of `ManifestWork`. Note that the "resourceNames"
+     should also be specified.
+3. The work-agent writes the objects to the managed cluster if SAR passes.
+   3.1 Otherwise set the failure condition to the `ManifestWork`.
+4. The work-agent reports the results back to the hub cluster.
+
+### Proposing changes to the work agent
+
+
+### Backward Compatibility
+
+Considering the data-plane backward compatibility, those existing legacy 
+`ManifestWork` resources will be considered as nil executor subject, so
+the work agent keeps using the original service account.
+
+
+### (Optional) Authorizing hub-user to executor subjects
+
+We can do an optional authorization in the hub cluster to check whether the
+manipulator of the `ManifestWork` can use a `ServiceAccount` as the executor
+subject by the integration between `ManagedServiceAccount` api. The 
+authorization works by plugging an admission webhook to the hub cluster.
+The webhook list/watches the `ManagedServiceAccountBinding` resource which
```

Comment:
Could you actually use RBAC in the hub cluster to manage this based on a virtual verb (`useOnManagedCluster`?) permission?

Reply:
added a new section about authorizing hub users via SAR in an validating admission webhook